src/ucs/debug/log.c

```diff
@@ -296,7 +296,14 @@ ucs_log_default_handler(const char *file, unsigned line, const char *function,
         /* Add source file name to the hash */
         match  = fnmatch(ucs_global_opts.log_component.file_filter, file, 0) !=
                  FNM_NOMATCH;
-        khiter = kh_put(ucs_log_filter, &ucs_log_global_filter, file, &khret);
+
+        filename = ucs_strdup(file, "log filter filename");
```

Comment:
maybe replace it by ucs_string_set_t?

Reply:
1- Currently `ucs_log_global_filter` is used to cache if a given file is a match or not. That boolean is accessed with `kh_val()`. But with `ucs_string_set_t` we can only track presence, so we would need to add one set to track caching state (already cached or not), and another set to track matching value (true or false).

2- If suggestion is to add `ucs_string_set_t` to handle memory allocation/tracking only, seems it does not expose string pointers, seems the function could be added though.

3- IMHO seems simpler like it currently is, let me know which option or any other.

Reply:
ok